src/xenia/kernel/xevent.cc

```diff
@@ -70,5 +71,33 @@ int32_t XEvent::Reset() {
 
 void XEvent::Clear() { event_->Reset(); }
 
+bool XEvent::Save(ByteStream* stream) {
+  XELOGD("XEvent %.8X (%s)", handle(), manual_reset_ ? "manual" : "auto");
+  SaveObject(stream);
+
+  // TODO: Write if we're currently signaled.
```

Comment:
you'll have to fix up the event after you query - if it comes back that it was signalled you'll need to re-signal it. because you only do this when nothing else should be waiting on that it should be ok